### DISCUSSION TITLE
How to configure IModelJsExpressServer to allow real-time data sending, for example iot senser data?

### DISCUSSION THREAD

**Discussion by Snailya** [iTwin/itwinjs-core#3431] [Q&A] [answered]
I want to integrate sensor data display in my itwin app.
Currently, the app backend is configured using IModelJsExpressServer.
Is this able to allow socket connection? If so, how?

**Comment by ColinKerr** [comment, marked as answer]
IModelJsExpressServer wraps an express Application.  If you derive from it you can access the protected _app variable which holds the Application.  There are several Open Source packages that add web socket support to express.

If the IoT sensor data is already accessible from another source you could integrate it on the frontend and avoid having to route the data through the backend.
